bundles/org.openhab.binding.argoclima/src/main/java/org/openhab/binding/argoclima/internal/configuration/ArgoClimaConfigurationLocal.java

```diff
@@ -0,0 +1,222 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.argoclima.internal.configuration;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.argoclima.internal.ArgoClimaBindingConstants;
+import org.openhab.binding.argoclima.internal.exception.ArgoConfigurationException;
+
+/**
+ * The {@link ArgoClimaConfigurationLocal} class extends base configuration parameters with ones specific
+ * to local connection (including a remote API stub / proxy)
+ *
+ * @author Mateusz Bronk - Initial contribution
+ */
+@NonNullByDefault
+public class ArgoClimaConfigurationLocal extends ArgoClimaConfigurationBase {
+    public enum ConnectionMode {
+        LOCAL_CONNECTION,
+        REMOTE_API_STUB,
+        REMOTE_API_PROXY
+    }
+
+    public enum DeviceSidePasswordDisplayMode {
+        NEVER,
+        MASKED,
+        CLEARTEXT
+    }
+
+    /**
+     * Argo configuration parameters specific to local connection & API stub
+     * These names are defined in thing-types.xml and get injected on instantiation
+     * through {@link org.openhab.core.thing.binding.BaseThingHandler#getConfigAs getConfigAs}
+     */
+    private String hostname = "";
+    private ConnectionMode connectionMode = ConnectionMode.LOCAL_CONNECTION;
+    private int hvacListenPort = 1001;
+    private String localDeviceIP = "";
+    private boolean useLocalConnection = true;
+    private int stubServerPort = 8239; // Note the original Argo server listens on '80', but picking a non privileged
+                                       // port (>1024) as a default, since this needs remapping on firewall, and openHAB
+                                       // is typically listening on 80 or 8080
+    private List<String> stubServerListenAddresses = List.of("0.0.0.0");
+    private DeviceSidePasswordDisplayMode includeDeviceSidePasswordsInProperties = DeviceSidePasswordDisplayMode.NEVER;
+    private boolean matchAnyIncomingDeviceIp = false;
+
+    /**
+     * Retrieves the *target* IP address of the LOCAL Argo device (from hostname and/or IP)
+     *
+     * @return The IP address of the Argo device (for use in local communication)
+     * @throws ArgoConfigurationException if no IP address for the {@code hostname} could be found
+     */
+    public InetAddress getHostname() throws ArgoConfigurationException {
```

Comment:
These methods can be extracted out to a static utility class 
getHostname()
getLocalDeviceIP()
getStubServerListenAddresses()

Reply:
I started to apply these changes, but reverted upon realizing the only thing I'd really be extracting is a 1:1 **passtrough** to `InetAddress.getByName()`.
_There are multiple lines, but upon 2nd read, these methods don't do anything more than passing through to `getByName` and throwing context-specific exceptions._ _I think the latter shouldn't be extracted to utils, as it belongs to the configuration parameter context, not anywhere global._

Please take a look if it still makes sense to change it. For now, I'm planning to leave them as-is (don't see value-add to extract).